script.service.tvheadend-route-latencies/default.py

```diff
@@ -0,0 +1,163 @@
+    return latency
+
+#Tries to download the main page ("/") from the server with the given url (should be in "hostname[:port]" format)
+# - every 10 seconds and no longer than sec seconds.
+#Returns string: average round trip time in localized format: "dd.d ms[, dd% packets loss]"
+#                or localized string "no connection" if there was no response.
+#It works by analogy to ping(hostname, sec). I used it once when I couldn't use ping.
+def http_rtt(url, sec):
+    if re.match('^[a-zA-Z0-9.:-]{1,253}$', url) is None: return lang(32129)  #"incorrect address format"
+    url = 'http://' + url
+    wait_sec = 10                                  #number of seconds to wait between requests
+    total_requests = 1 + sec/wait_sec              #number of requests that will be sent
+    total_answers  = 0
+    packet_loss    = 0
+    total_time     = 0
+    s = requests.Session()
+    try:
+        s.get(url, timeout=5).text                 #make a TCP connection to speed up proper requests
+    except requests.RequestException:
+        pass
+    time.sleep(1)                                  #be nice
+    for i in range(0, total_requests):
+        try:
+            start = timer()
+            s.get(url, timeout=5).text             #thanks to .text the session can be closed when not needed
+            total_time += timer() - start
+            total_answers += 1
+            if i != total_requests - 1: time.sleep(wait_sec)  #wait <wait_sec> seconds
+        except requests.RequestException:          #the web page could not be downloaded in <timeout> seconds
+            packet_loss += 1
+    if total_answers == 0:
+        latency = lang(32130)  #"no connection"
+    else:
+        latency = format(total_time*1000/total_answers, '.1f') + " ms" #convert seconds to ms, calculate average, round to 1 decimal place
+        if packet_loss != 0:
+            latency += ", " + str(packet_loss*100/total_requests) + "% " + lang(32131)  #"packets loss"
+    return latency
+
+#Shows latency (RTT) and packets loss to consecutive steps on the way from Kodi to Tvheadend server.
+#For more info see description in the README.md file.
+class TVNetworkHealthWindow(pyxbmct.AddonDialogWindow):
+    def setAnimation(self, control):
+        control.setAnimations([('WindowOpen',  'effect=fade start=0   end=100 time=300',),
+                               ('WindowClose', 'effect=fade start=100 end=0   time=300',)])
+
+    def __init__(self):
+        super(TVNetworkHealthWindow, self).__init__("Tvheadend route latencies")
+        try:  #comment it temporary to see detailed log in case of errors
+            self.testlength = int(ADDON.getSetting("testlength"))  #how long a single test will take
+
+            #--------------- IP addresses of consecutive routers and servers ---------------
+            local_router    = xbmc.getInfoLabel('Network.GatewayAddress')
```

Comment:
Platform and version specific wheels for `netifaces` suggest that it has binary components so it is not acceptable for a Kodi addon. But `Network.GatewayAddress` limitation is not exactly a show stopper, and the rest looks good.